utils/cp_parallel.py

```diff
@@ -0,0 +1,155 @@
+            logging.Formatter("[%(asctime)s] [Process ID: %(process)d] %(message)s")
+        )
+
+        # add the file handler to the logger
+        logger.addHandler(file_handler)
+
+        # log plate name and output string
+        logger.info(f"Plate Name: {plate_name}")
+        logger.info(f"Output String: {output_string}")
+
+        # remove the file handler to avoid duplication in the next loop iteration
+        logger.removeHandler(file_handler)
+
+
+def run_cellprofiler_parallel(
+    plate_info_dictionary: dict,
+    run_name: str,
+) -> None:
+    """
+    This function utilizes multi-processing to run CellProfiler pipelines in parallel.
+
+    Args:
+        plate_info_dictionary (dict): dictionary with all paths for CellProfiler to run a pipeline
+        run_name (str): a given name for the type of CellProfiler run being done on the plates (example: whole image features)
+
+    Raises:
+        FileNotFoundError: if paths to pipeline and images do not exist
+    """
+    # create a list of commands for each plate with their respective log file
+    commands = []
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
+
+    # iterate through each plate in the dictionary
+    for _, info in plate_info_dictionary.items():
+        # set paths for CellProfiler
+        path_to_pipeline = info["path_to_pipeline"]
+        path_to_images = info["path_to_images"]
+        path_to_output = info["path_to_output"]
+
+        # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+        if not pathlib.Path(path_to_pipeline).resolve(strict=True):
+            raise FileNotFoundError(
+                f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist"
+            )
+        if not pathlib.Path(path_to_images).is_dir():
+            raise FileNotFoundError(
+                f"Directory '{pathlib.Path(path_to_images).name}' does not exist or is not a directory"
+            )
+        # make output directory if it is not already created
+        pathlib.Path(path_to_output).mkdir(exist_ok=True)
+
+        # creates a command for each plate in the list
+        command = [
+            "cellprofiler",
+            "-c",
+            "-r",
+            "-p",
+            path_to_pipeline,
+            "-o",
+            path_to_output,
+            "-i",
+            path_to_images,
+        ]
+        # creates a list of commands
+        commands.append(command)
+
+    # set the number of CPUs/workers as the number of commands
+    num_processes = len(commands)
+
+    # make sure that the number of workers does not exceed the maximum number of workers for the machine
+    if num_processes > multiprocessing.cpu_count():
+        raise MaxWorkerError(
+            "Exception occurred: The number of commands exceeds the number of CPUs/workers. Please reduce the number of commands."
+        )
```

Comment:
Instead of throwing an error consider using logic to scale to the number processes that are less than that number of processes. 

Pseudo code:

num_of_jobs = len(commands)
if not num_of_jobs >= multiprocessing.cpu_count():
    # run cp parallel here
else:
    # run n number of jobs at a time where n is equal to the number of cores - 2

Reply:
Hmm interesting point! I think we can include this as an issue for the CP_Parallel repo since I can also see a reason to throw an error instead of just scaling in case someone would prefer to have a chance to change their function or plates they are processing. For now, given the scale of this PR, I will leave it.